packages/common/src/dedupe.ts

```diff
@@ -0,0 +1,13 @@
+export const dedupe = <item, id>(
```

Comment:
Maybe use a function expression to be consistent? Also a docstring would be helpful considering there's some complexity around behavior of objects vs primitives and the `getId` option.